### PR TITLE
[aptos-framework] Cleanup Property map

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Cleans up some docs, fixes some wrong error codes, reordered errors to have some consistency.

Also, made some functions inline that were not used anywhere else, removed duplicate code.

### Test Plan
`aptos move test`

Additionally, I ran coverage.  For some reason `source` mode bugs out, but there are the metrics.  Will likely want to add a test for the untested ones before it goes to prod.

```
+-------------------------+
| Move Coverage Summary   |
+-------------------------+
Module 0000000000000000000000000000000000000000000000000000000000000004::property_map
        fun add
                total: 221
                covered: 199
                % coverage: 90.05
        fun add_typed
                total: 136
                covered: 123
                % coverage: 90.44
        fun burn
                total: 6
                covered: 0
                % coverage: 0.00
        fun contains_key
                total: 19
                covered: 0
                % coverage: 0.00
        fun generate_mutator_ref
                total: 4
                covered: 4
                % coverage: 100.00
        fun init
                total: 7
                covered: 7
                % coverage: 100.00
        fun length
                total: 16
                covered: 11
                % coverage: 68.75
        fun prepare_input
                total: 257
                covered: 238
                % coverage: 92.61
        fun read
                total: 129
                covered: 113
                % coverage: 87.60
        fun read_address
                total: 14
                covered: 0
                % coverage: 0.00
        fun read_bool
                total: 14
                covered: 11
                % coverage: 78.57
        fun read_bytes
                total: 14
                covered: 11
                % coverage: 78.57
        fun read_string
                total: 14
                covered: 11
                % coverage: 78.57
        fun read_u128
                total: 14
                covered: 11
                % coverage: 78.57
        fun read_u16
                total: 14
                covered: 11
                % coverage: 78.57
        fun read_u256
                total: 14
                covered: 11
                % coverage: 78.57
        fun read_u32
                total: 14
                covered: 11
                % coverage: 78.57
        fun read_u64
                total: 14
                covered: 11
                % coverage: 78.57
        fun read_u8
                total: 14
                covered: 14
                % coverage: 100.00
        fun remove
                total: 23
                covered: 16
                % coverage: 69.57
        fun update
                total: 230
                covered: 202
                % coverage: 87.83
        fun update_typed
                total: 143
                covered: 126
                % coverage: 88.11
>>> % Module coverage: 85.73
+-------------------------+
| % Move Coverage: 85.73  |
+-------------------------+

```

### CODE REVIEW

**Review by movekevin — APPROVED** [review]
Thank you ser! This is so much better!

File: aptos-move/framework/aptos-token-objects/doc/token.md [hunk 1]
```diff
@@ -188,34 +102,6 @@ directly understand the behavior in a writeset.
 
 
 
-<details>
-<summary>Fields</summary>
-
-
-<dl>
-<dt>
-<code>mutated_field_name: <a href="../../aptos-framework/../aptos-stdlib/../move-stdlib/doc/string.md#0x1_string_String">string::String</a></code>
```
Comment (movekevin) [inline aptos-move/framework/aptos-token-objects/doc/token.md:197]: Do you know why these changed without changes in the corresponding Move files?

Comment (gregnazario) [reply aptos-move/framework/aptos-token-objects/doc/token.md]: Fixed, I accidentally uploaded the wrong ones

File: aptos-move/framework/aptos-token-objects/sources/property_map.move [hunk 2]
```diff
@@ -215,34 +201,36 @@ module aptos_token_objects::property_map {
...
-            error::not_found(EPROPERTY_MAP_DOES_NOT_EXIST),
-        );
+        check_property_map_exists(object::object_address(object));
         let property_map = borrow_global<PropertyMap>(object::object_address(object));
         let property_value = simple_map::borrow(&property_map.inner, key);
         let new_type = to_external_type(property_value.type);
         (new_type, property_value.value)
     }
 
+    inline fun check_property_map_exists(object: address) {
```
Comment (davidiw) [inline aptos-move/framework/aptos-token-objects/sources/property_map.move:226]: in the interest of being anal, we are already in `property_map` so the additional namespacing doesn't add value, we could just consider
`assert_exists`?

check feels more like something I'd say should return a bool

**Review by davidiw — APPROVED** [review]

**Pull request merged**